File: Session 2 - The Art of Prompt Engineering/Lab 3 - GitHub Copilot/5 Code Refactoring.py
def calculate_area_and_perimeter(shape, dimensions):
    if shape == 'rectangle':
        length, width = dimensions
        area = length * width
        perimeter = 2 * (length + width)
        return area, perimeter
    elif shape == 'circle':
        radius = dimensions[0]
        area = 3.14159 * radius * radius
        perimeter = 2 * 3.14159 * radius
        return area, perimeter
    elif shape == 'triangle':
        base, height = dimensions
        area = 0.5 * base * height
        perimeter = 'Not implemented'
        return area, perimeter
    else:
        return 'Invalid shape', 'Invalid shape'

# Example usage
print(calculate_area_and_perimeter('rectangle', (5, 10)))
print(calculate_area_and_perimeter('circle', (7,)))
print(calculate_area_and_perimeter('triangle', (6, 8)))
print(calculate_area_and_perimeter('hexagon', (6,)))

File: Session 2 - The Art of Prompt Engineering/Lab 3 - GitHub Copilot/5 Code Refactoring.cs
using System;

namespace ShapeAreaCalculator
{
    public class ShapeCalculator
    {
        public double CalculateCircleArea(double radius)
        {
            return Math.PI * radius * radius;
        }

        public double CalculateRectangleArea(double length, double width)
        {
            return length * width;
        }

        public double CalculateTriangleArea(double baseLength, double height)
        {
            return 0.5 * baseLength * height;
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            ShapeCalculator calculator = new ShapeCalculator();

            double circleArea = calculator.CalculateCircleArea(5);
            Console.WriteLine($"Circle Area: {circleArea}");

            double rectangleArea = calculator.CalculateRectangleArea(4, 6);
            Console.WriteLine($"Rectangle Area: {rectangleArea}");

            double triangleArea = calculator.CalculateTriangleArea(3, 7);
            Console.WriteLine($"Triangle Area: {triangleArea}");
        }
    }
}